dataflow_test.py

```diff
@@ -523,3 +523,71 @@ def test_count_filter_counts_correctly():
 
     assert result.n_passed == len( odd_numbers)
     assert result.n_failed == len(even_numbers)
+
+
+@parametrize("component",
+             (df.map   (lambda x: x)    ,
+              df.filter(lambda x: x > 0),
+              df.sink  (print)          ,
+              df.branch(df.sink(print)) ,
+              df.pipe  (df.map(abs))    ))
+def test_string_to_pick_ignores_components(component):
+    assert component is df.string_to_pick(component)
+
+
+def test_string_to_pick():
+
+    # string_to_pick creates a pipe component that picks
+    # an item from the namespace and pushes it through the pipe
+
+    the_source_elements = list(range(10))
+    the_source          = (dict(x=i) for i in the_source_elements)
```

Comment:
I'd be tempted to populate it with more attributes than just the one that you are going to extract.